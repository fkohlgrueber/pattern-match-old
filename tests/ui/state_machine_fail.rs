#[deny(state_machine)]

struct Phone;

impl Phone {
    fn begin_incoming_call(&mut self) {
        unimplemented!()
    }
    fn accept_call(&mut self) {
        unimplemented!()
    }
    fn end_call(&mut self) {
        unimplemented!()
    }
    fn begin_outgoing_call(&mut self) {
        unimplemented!()
    }
    fn finished_dialing(&mut self) {
        unimplemented!()
    }
    fn call_accepted(&mut self) {
        unimplemented!()
    }
}

fn main() {
    let mut x = Phone;
    x.begin_incoming_call();
    x.end_call(); //~ ERROR not applicable for states {"Ringing"}
}
