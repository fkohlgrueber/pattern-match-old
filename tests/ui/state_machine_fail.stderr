error: state transition `end_call` not applicable for states {"Ringing"}
  --> $DIR/state_machine_fail.rs:29:5
   |
29 |     x.end_call(); //~ ERROR not applicable for states {"Ringing"}
   |     ^^^^^^^^^^^^
   |
   = note: #[forbid(state_machine)] on by default

error: aborting due to previous error

